### DISCUSSION TITLE
Snowflake - last_query_id overwritten by DBeaver

### DISCUSSION THREAD

**Discussion by chuckh1958** [dbeaver/dbeaver#15409] [Q&A]
When using the snowflake driver and last_query_id() function to return the results of the last query, you get the results of an internal query run by dbeaver to get the current database and schema, not the results of the last query executed by the user. Is there a way to prevent this or is this a bug that needs to be fixed?

Example:
show databaes;

select *
from table(result_scan(last_query_id()));

This should show the results of show databases again but it does not.

**Comment by LonwoLonwo** [comment]
Hello @chuckh1958 

Maybe disabling this setting can help you - It will turn off a search of the current database and schema after query execution in the SQL Editor.

![2022-02-09 00_35_44-Preferences](https://user-images.githubusercontent.com/45152336/153079705-09e8d8cb-0773-4202-842e-3ba3692a166f.png)

**Comment by chuckh1958** [comment]
Nope. Didn’t change anything.

**Comment by LonwoLonwo** [comment]
Finally, I found it.

You can disable this behavior with this setting:
Preferences -> Editors -> SQL Editor -> SQL Processing -> "Refresh active schema after SQL execution"

![2022-03-10 13_04_24-DBeaver Ultimate 22 0 0 - _SNOWFLAKE_SAMPLE_DATA_ Script-28](https://user-images.githubusercontent.com/45152336/157639203-974b6e6a-3c19-430e-8c6a-2e95aeb31c64.png)
